ch02.adoc

```diff
@@ -268,3 +268,213 @@ They may not be attached to a group, even if all variables within that group use
+Fragment data shape `(17, 45, 180)` +
+`17` vertical levels +
+`(0, -45]` degrees north +
+`[180, 360)` degrees east
+
+| *Fragment array position `[0, 2, 0]`*
+
+Fragment file name `file_E.nc` +
+Fragment data shape `(17, 45, 180)` +
+`17` vertical levels +
+`(-45, -90]` degrees north +
+`[0, 180)` degrees east | *Fragment array position `[0, 2, 1]`*
+
+Fragment file name `file_F.nc` +
+Fragment data shape `(17, 45, 180)` +
+`17` vertical levels +
+`(-45, -90]` degrees north +
+`[180, 360)` degrees east
+|===============
+Six fragments are arranged in a three-dimensional fragment array with shape `(1, 3, 2)`.
+Each fragment spans the entirety of the Z dimension, but only a part of the Y-X plane, which has 1 degree resolution.
+The fragments combine to create three-dimensional aggregated data that have global `(Z, Y, X)` coverage, with shape `(17, 181, 360)`.
+The Z aggregated dimension is spanned by 1 fragment, the Y aggregated dimension is spanned by 3 fragments, and the X aggregated dimension is spanned by 2 fragments.
+See <<example-L.4>> for a CDL representation of this fragment array.
+====
+
+The fragment array must be defined by an aggregation variable's **`aggregated_data`** attribute.
+This attribute takes a string value comprising blank-separated elements of the form "__feature: variable__", where __feature__ is a case-sensitive keyword that identifies a feature of the fragment array, and __variable__ is a __fragment array variable__ that provides the feature's values for each fragment in the fragment array.
+The order of elements in the **`aggregated_data`** attribute is not significant.
+
+There are four standardized and mandatory features, given by the `file`, `format`, `address`, and `shape` keywords; and any amount of non-standardized features are also allowed:
+
+*file*
+
+The string-valued `file` fragment array variable defines the locations of the fragment files.
+In general its dimensions correspond to, and have the same sizes as, the fragment array dimensions in the same order as they appear in the conceptual fragment array.
+The  `file` fragment array variable values provide the fragment file names.
+Each fragment file name must be a Uniform Resource Identifier (URI) <<URI>> that is either an __absolute URI__ (a URI that begins with a scheme component followed by a `:` character, such as `\file://data/file.nc`, `\https://data/file.nc`, and `s3://data/file.nc`), or else a __relative-path URI reference__ (a URI that is not an absolute URI and which does not begin with a `/` or `#` character, such as `file.nc`, `../file.nc`, and `data/file.nc`).
+A relative-path URI reference is taken as being relative to the current location of the aggregation file.
+If the aggregation file is moved to another location, then a fragment file identified by an absolute URI will still be accessible, whereas a fragment file identified by a relative-path URI reference will also need be moved to preserve the relative reference.
```

Comment:
Hi @bnlawrence  - could you possibly review these three lines?

The terms _absolute URI_ and _relative-path URI reference_ are well defined in the [URI spec](https://www.rfc-editor.org/rfc/rfc3986.html) (which is the target of the `<<URI>>` reference). My definitions in parentheses are, I hope, a more consumable distillation of how the URI spec defines them.

Many thanks,
David

Reply:
I wonder whether you might want to say something like:
> a URI that begins with a scheme component followed by a `:` character, such as `\file://data/file.nc`, `\https://remote.host/data/file.nc`, `s3://remote.host/data/file.nc`,  or`locally_meaningful_protocol://UID`) or else ...)

I would also consider adding the following line. 

Where relative-URI paths are used, data producers are strongly encouraged to use the identifier mechanism to help discriminate between possible targets of lost relative references.

Reply:
Thanks, Bryan - what is the "identifier mechanism"?

Reply:
All resolved offline. Text soon to be updated.